src/librustc_typeck/check/writeback.rs

```diff
@@ -399,7 +399,7 @@ impl<'cx, 'tcx> WritebackCx<'cx, 'tcx> {
         assert_eq!(fcx_tables.hir_owner, self.tables.hir_owner);
 
         for (&def_id, c_sig) in fcx_tables.user_provided_sigs.iter() {
-            if cfg!(debug_assertions) && c_sig.has_local_value() {
+            if cfg!(debug_assertions) && c_sig.needs_infer() {
                 span_bug!(
                     self.fcx.tcx.hir().span_if_local(def_id).unwrap(),
                     "writeback: `{:?}` is a local value",
```

Comment:
Same here (check the entire file, or maybe even the entire compiler).

Reply:
jup